core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -168,7 +168,42 @@ private[spark] object PythonRDD extends Logging {
   }
 
   def toLocalIteratorAndServe[T](rdd: RDD[T]): Array[Any] = {
-    serveIterator(rdd.toLocalIterator, s"serve toLocalIterator")
```

Comment:
No, the remaining jobs are not triggered. The python iterator finishes consuming the data from the current job, then sends a command for Scala iterator to stop.

Reply:
How about previous behavior? The behavior before will trigger them? Looks like `toLocalIterator` won't trigger the job if we don't iterate the data on a partition.

Reply:
The previous behavior was that the Scala local iterator would advance as long as the `write` calls to the socket are not blocked. So this means when Python reads a batch (auto-batched elements) from the current partition, this will unblock the Scala call to write and could start the next job.

Reply:
Once the local iterator at Python side is out of scope and so the iterator is not fully consumed, will it block the write call at Scala? Seems to me that it will and we shouldn't see unneeded jobs to be triggered after that, doesn't?

Reply:
The previous behavior is when the iterator goes out of scope, the socket eventually is closed. This creates the error on the Scala side and the writing thread is terminated, so no more jobs are triggered but the user sees this error.